packages/@lwc/engine-server/src/serializer.ts

```diff
@@ -18,12 +25,23 @@ function serializeAttributes(attributes: HostAttribute[]): string {
         .join(' ');
 }
 
+function serializeComment(commentNode: HostComment) {
+    const { parts, dynamicIndexes } = commentNode;
+
+    const escapedParts = parts.map((part) => (part == undefined ? '' : String(part)));
```

Comment:
Did you mean `part === undefined` or `part == null`? Or maybe `String(part ?? '')`?

Reply:
i meant `part == undefined`, which is the same as `part == null`, right? is there a preference for it? i don't mind to change it to `part == null` or even `part === null || part === undefined`.

Reply:
Yeah, you're right:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness#loose_equality_using

To be honest, I dislike this pattern because enough time always passes before the next time I see it, that I have to look up what it means. I prefer `String(part ?? '')` but if you prefer using `==` then let's change it from `undefined` to `null` since the latter follows convention. There's even an eslint rule exception explicitly for `null`: https://eslint.org/docs/rules/eqeqeq

Reply:
yeah, `String(part ?? '')` is better, though in order to use it, we should [drop support for node versions](https://github.com/salesforce/lwc/blob/master/package.json#L74) lower than 14 (iirc). if we don't want to do it, I can use `part === null || part === undefined`

Reply:
@jodarove keep in mind that `tsc` is processing the code after you. As configured today, the `??` will be transformed for you. No need to worry about using such syntax in typescript files.